elasticsearch/includes/classes/class-health-job.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace Automattic\VIP\Elasticsearch;
+
+use Automattic\VIP\Elasticsearch\Health as Health;
+
+class HealthJob {
+
+	/**
+	 * The name of the scheduled cron event to run the health check
+	 */
+	const CRON_EVENT_NAME = 'vip_search_healthcheck';
+
+	/**
+	 * Custom cron interval name
+	 */
+	const CRON_INTERVAL_NAME = 'vip_search_healthcheck_interval';
+
+	/**
+	 * Custom cron interval value
+	 *
+	 * 30 minutes in seconds
+	 */
+	const CRON_INTERVAL = 60 * 30;
+
+	/**
+	 * Initialize the job class
+	 *
+	 * @access	public
+	 */
+	public function init() {
+		// Add the custom cron schedule
+		add_filter( 'cron_schedules', [ $this, 'filter_cron_schedules' ], 10, 1 );
+
+		$this->schedule_job();
+	}
+
+	/**
+	 * Schedule health check job
+	 *
+	 * Add the event name to WP cron schedule and then add the action
+	 */
+	public function schedule_job() {
+		if ( ! wp_next_scheduled( self::CRON_EVENT_NAME )  ) {
+			wp_schedule_event( time(), self::CRON_INTERVAL_NAME, self::CRON_EVENT_NAME );
+		}
+
+		add_action( self::CRON_EVENT_NAME, [ $this, 'check_health' ] );
+	}
+
+	/**
+	 * Filter `cron_schedules` output
+	 *
+	 * Add the custom interval to WP cron schedule
+	 *
+	 * @param		array	$schedule
+	 *
+	 * @return	mixed
+	 */
+	public function filter_cron_schedules( $schedule ) {
+		if ( isset( $schedule[ self::CRON_INTERVAL_NAME ] ) ) {
+			return $schedule;
+		}
+
+		// Not actually five minutes; we want it to run faster though to get through everything.
+		$schedule[ self::CRON_INTERVAL_NAME ] = [
+			'interval' => self::CRON_INTERVAL,
+			'display' => __( 'VIP Search Healthcheck time interval' ),
+		];
+
+		return $schedule;
+	}
+
+	/**
+	 * Check ElasticSearch index health
+	 */
+	public function check_health() {
+		$user_results = Health::validate_index_users_count();
+
+
+		$this->process_results( $user_results );
+
+		$post_results = Health::validate_index_posts_count();
```

Comment:
Aside: I wonder if in the `Health` class we should just have a single entry point that runs the various validations (`Health::validate()`)? That way if (and when) we add additional checks (e.g. cluster health, index exists, etc.), they're automatically checked here as well.

Not blocking, but definitely something we should consider.